hack/tools/release/notes.go

```diff
@@ -152,8 +153,18 @@ func run() int {
 			commits[last].merge = line
 			continue
 		case line == "":
+		case line == "```release-note":
```

Comment:
I think this tool is currently only parsing the commit message. But it seems like we add the release-note to the PR description not to the commit message.

I would have expected us to call the GitHub API to retrieve the PR description of every commit to be able to handle the release note in the PR descrption

(Maybe I'm missing something)

Reply:
When we change from reading from the PR description, let's make sure to always add the PR number like we do in the other lists e.g. - some test (#8091)